File: ina_2sem/labor_cs/3_lab/1_task/Pieciokat.java
public class Pieciokat extends Figura {
    private final double bok;

    public Pieciokat(double bok) {
        if (bok <= 0) throw new IllegalArgumentException("Bok musi być dodatni.");
        this.bok = bok;
    }

    @Override
    public double obliczPole() {
        return (5 * bok * bok) / (4 * Math.tan(Math.PI / 5));
    }

    @Override
    public double obliczObwod() {
        return 5 * bok;
    }

    @Override
    public String podajNazwe() {
        return "Pięciokąt foremny";
    }
}

File: ina_2sem/labor_cs/3_lab/2_task/Pieciokat.cpp
#include "Pieciokat.h"

Pieciokat::Pieciokat(double b) : bok(b) {
    if (b <= 0) throw std::invalid_argument("Bok musi być dodatni.");
}

double Pieciokat::obliczPole() {
    return (5 * bok * bok) / (4 * tan(M_PI / 5));
}

double Pieciokat::obliczObwod() {
    return 5 * bok;
}

std::string Pieciokat::podajNazwe() {
    return "Pięciokąt foremny";
}
